src/code_6000.c

```diff
@@ -50,16 +48,25 @@ s32 heap_free(s32 size) {
     }
 }
 
-INCLUDE_ASM("code_6000", collision_heap_create);
+s32 collision_heap_create(void) {
+    if (_heap_create(&D_80268000, 0x18000) == -1) {
+        return -1;
+    }
+    return 0;
+}
 
-INCLUDE_ASM("code_6000", collision_heap_malloc);
+s32 collision_heap_malloc(s32 size) {
+    if (!(*gGameStatusPtr)->isBattle) {
+        return _heap_malloc(&D_80268000, size);
+    } else {
+        return _heap_malloc(&D_803DA800, size);
+    }
+}
 
-#ifdef NON_MATCHING
 s32 collision_heap_free(s32 size) {
```

Comment:
Is this really size? Looking up a heap entry just via size seems uhhh, well, suspect to say the least. Usually free takes in a  pointer and not a size. (Also seems like _heap_free can handle multiple heaps so you need to tell it which one, hence the first ptr). I'd make the arg to this func take in a ``void *`` to a var called `buffer` or `buf` or so.

Reply:
Oh yeah oops it's a pointer, not a size. clover calls it "data" which makes sense to me. idk

Reply:
Data sounds like a fine name to me